### PR TITLE
DOC-2429 release/7.2: Update Type Functions page

### PR DESCRIPTION
**Pull request by franrob-projects**
Adds examples, headings and 'missing' note.

### CODE REVIEW

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 1]
```diff
@@ -1,36 +1,225 @@
 = Type Functions
-:description: Type functions perform operations that check or convert expressions.
+:description: The following checks if the supplied expression is a (data type).
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc]: My bad, I meant use this text as the basis for the **Description** section in each of the type-checking functions.

```suggestion
:description: Type functions perform operations that check or convert expressions.
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 2]
```diff
@@ -1,36 +1,225 @@
...
 
 [abstract]
 {description}
 
+NOTE: If any arguments to any of the following functions are [.out]`MISSING` then the result is also [.out]`MISSING` (i.e.
+no result is returned).
+Similarly, if any of the arguments passed to the functions are `NULL` or are of the wrong type (e.g.
+an integer instead of a string), then `NULL` is returned as the result.
+
 == Type-Checking Functions
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc]: This is an interesting one. None of the other function pages have sub-headings for the different types of function. I wouldn't rule it out for the future. But for now, delete the two sub-headings **Type-Checking Functions** and **Type-Conversion Functions**, for consistency with other function pages.

```suggestion
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 3]
```diff
@@ -1,36 +1,225 @@
...
 
-ISARRAY(expression)
+== ISARRAY(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:15]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isarray]
== ISARRAY(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 4]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:17]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is an array.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 5]
```diff
@@ -1,36 +1,225 @@
...
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISATOM(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:58]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isatom]
== ISATOM(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 6]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:60]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a Boolean, a number, or a string.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 7]
```diff
@@ -1,36 +1,225 @@
...
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISBOOLEAN(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:92]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isboolean]
== ISBOOLEAN(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 8]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:94]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a Boolean.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 9]
```diff
@@ -1,36 +1,225 @@
...
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:88]: Wrong results. Replace this with the results of the ISATOM query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 10]
```diff
@@ -1,36 +1,225 @@
...
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:122]: Wrong results. Replace with the results of the ISBOOLEAN query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 11]
```diff
@@ -1,36 +1,225 @@
...
+----
+====
+
+== ISNUMBER(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:126]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isnumber]
== ISNUMBER(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 12]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:128]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a number.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 13]
```diff
@@ -1,36 +1,225 @@
...
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:156]: Wrong results. Replace with the results of the ISNUMBER query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 14]
```diff
@@ -1,36 +1,225 @@
...
+----
+====
+
+== ISOBJECT(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:160]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isobject]
== ISOBJECT(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 15]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:162]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is an object.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 16]
```diff
@@ -1,36 +1,225 @@
...
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:190]: Wrong results. Replace with the results of the ISOBJECT query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 17]
```diff
@@ -1,36 +1,225 @@
...
+----
+====
+
 ISSTRING(expression)
 
 Returns True if expression is a string, otherwise returns MISSING, NULL or false.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:196]: Delete repeated text

```suggestion
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 18]
```diff
@@ -1,36 +1,225 @@
...
 
+== ISSTRING(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:198]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-isstring]
== ISSTRING(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 19]
```diff
@@ -1,36 +1,225 @@
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:200]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a string.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 20]
```diff
@@ -1,36 +1,225 @@
...
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:228]: Wrong results. Replace with the results of the ISSTRING query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 21]
```diff
@@ -1,36 +1,225 @@
...
+----
+====
+
 TYPE(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:232]: Delete repeated text (as far as the TYPE heading below)

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 22]
```diff
@@ -44,7 +233,51 @@ Returns one of the following strings, based on the value of expression:
 * "object"
 * "binary"
 
+== TYPE(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:246]: Needs unique anchor, as per functions template

```suggestion
[#fn-type-type]
== TYPE(expression)
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 23]
```diff
@@ -44,7 +233,51 @@ Returns one of the following strings, based on the value of expression:
...
+
+=== Description
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:248]: Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks the type of the supplied expression.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 24]
```diff
@@ -44,7 +233,51 @@ Returns one of the following strings, based on the value of expression:
...
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:293]: Wrong results. Replace with the results of the TYPE query.

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 25]
```diff
@@ -44,7 +233,51 @@ Returns one of the following strings, based on the value of expression:
...
+----
+====
+
 == Type-Conversion Functions
+:description: The following converts the supplied expression to a (data type).
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:298]: Delete the **Type-Conversion Functions** heading as discussed above. Also delete this `description` attribute &mdash; I meant that you should use this text as the basis for the Description section for each of the type-conversion functions.

```suggestion
```

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc [hunk 26]
```diff
@@ -44,7 +233,51 @@ Returns one of the following strings, based on the value of expression:
...
 
 TOARRAY(expression)
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/typefun.adoc:300]: Do the same changes (as described above) for each of the type-conversion functions.

**Review by simon-dew — CHANGES_REQUESTED** [review]
Good start 👍  &mdash; but still some way to go on this one

**Comment by franrob-projects** [comment]
> 

Thanks, and all the above noted. I know exactly how to amend, and will execute the changes (for backports)

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc (hunk 24 above)
Comment (franrob-projects) [reply modules/n1ql/pages/n1ql-language-reference/typefun.adoc:293]: Ahh ok apologies, this one needs a 'missing'. I will end in the next PR

File: modules/n1ql/pages/n1ql-language-reference/typefun.adoc (hunk 25 above)
Comment (franrob-projects) [reply modules/n1ql/pages/n1ql-language-reference/typefun.adoc:298]: Ahh ok, I get you. I was 50/50 on what to do there. Thanks

**Pull request merged**